core/interfaces/branch.py

```diff
@@ -156,17 +157,20 @@ def render_branch_list(self, remote_name=None, branches=None):
         if not branches:
             branches = [branch for branch in self._branches if not branch.is_remote]
 
+        current_worktree = self.get_current_worktree()
```

Comment:
This is a side-effect and as such not allowed in the render phase where you should just format the data you got.  It is also a redundant call because you have already all the branches in `branches`.